### PR TITLE
IDO changes

### PR DESCRIPTION
**Pull request by louist103**
Add support for arm64, and recompile IDO x86 in -O2 mode

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -79,9 +84,9 @@ CFLAGS += -G 0 -non_shared -Xfullwarn -Xcpluscomm -Iinclude -Isrc -Wab,-r4300_mu
 ifeq ($(shell getconf LONG_BIT), 32)
   # Work around memory allocation bug in QEMU
   export QEMU_GUEST_BASE := 1
-else
-  # Ensure that gcc treats the code as 32-bit
-  CC_CHECK += -m32
+else ($(DETECTED_ARCH),x86_64)
```
Comment (ethteck) [inline Makefile:87]: Not sure about this line. Other than that, lgtm!

**Comment by louist103** [comment]
Just an update there is still an issue when building on ARM so dont merge this yet

**Comment by fig02** [comment]
if we are going to go the route of hosting ido recompiled in a separate repository, can this be closed?

**Comment by ethteck** [comment]
I think it should be, yeah. Do you mind if we do @louist103 ?

**Comment by louist103** [comment]
Yeah I’ll close it. Since there are going to be other changes to this

**Pull request closed without merging**